projects/noloan2/app/src/main/java/hasadna/noloan2/LawsuitPdfActivity.java

```diff
@@ -0,0 +1,234 @@
+        lawsuit = lawsuit.replace("&lt;userFullName&gt;", userPrivateName + " " + userLastName);
+        lawsuit = lawsuit.replace("&lt;userId&gt;", userID);
+        lawsuit = lawsuit.replace("&lt;userAddress&gt;", userAddress);
+        lawsuit = lawsuit.replace("&lt;userPhone&gt;", userPhone);
+        lawsuit = lawsuit.replace("&lt;userFax&gt;", userFax);
+
+        // First spam company
+        lawsuit = lawsuit.replace("&lt;companyName&gt;", companyName);
+        lawsuit = lawsuit.replace("&lt;companyId&gt;", companyId);
+        lawsuit = lawsuit.replace("&lt;companyAddress&gt;", companyAddress);
+        lawsuit = lawsuit.replace("&lt;companyPhone&gt;", companyPhone);
+        lawsuit = lawsuit.replace("&lt;companyFax&gt;", companyFax);
+
+        // Second spam company
+        lawsuit = lawsuit.replace("&lt;company2Name&gt;", company2Name);
+        lawsuit = lawsuit.replace("&lt;company2Id&gt;", company2Id);
+        lawsuit = lawsuit.replace("&lt;company2Address&gt;", company2Address);
+        lawsuit = lawsuit.replace("&lt;company2Phone&gt;", company2Phone);
+        lawsuit = lawsuit.replace("&lt;company2Fax&gt;", company2Fax);
+
+        return lawsuit;
+    }
+
+    private String readTemplate() {
+
+        StringBuilder contentBuilder = new StringBuilder();
+        try {
+            BufferedReader in = new BufferedReader(new FileReader(lawsuitTemplateFilePath));
+            String str;
+            while ((str = in.readLine()) != null) {
+                contentBuilder.append(str);
+            }
+            in.close();
+        } catch (IOException e) {
+            Log.w(TAG, "Error reading lawsuit's template.");
+            e.printStackTrace();
+        }
+        return contentBuilder.toString();
+    }
+
+    private String getPdfFilename() {
+
+        return dateAndTimeFormatter.format(new Date()) + ".pdf";
+    }
+
+    private void initDirStructure() {
+
+        lawsuitMainPath = Paths.get(Environment.getExternalStorageDirectory().getPath(), getString(R.string.app_name), LAWSUIT_MAIN_DIR_NAME).toString();
+        initDir(lawsuitMainPath);
+
+        lawsuitTemplateDirPath = Paths.get(lawsuitMainPath, LAWSUIT_TEMPLATE_DIR_NAME).toString();
+        initDir(lawsuitTemplateDirPath);
+
+        // TODO: Currently template's file is been stored in user's device storage. Check to see if this should be part of the apk package.
+        try {
+            lawsuitTemplateFilePath = Paths.get(lawsuitTemplateDirPath, LAWSUIT_TEMPLATE_FILE_NAME).toString();
+        } catch (Exception e) {
+            Toast.makeText(this, "No template file found for the lawsuit.",
+                    Toast.LENGTH_LONG).show();
+        }
+
+        // Output
+        lawsuitOutputPath = Paths.get(lawsuitMainPath, LAWSUIT_OUTPUT_DIR_NAME).toString();
+        initDir(lawsuitOutputPath);
+    }
+
+    /**
+     * Create dir if it doesn't exists.
+     * @param path Directory to look for.
+     */
+    private void initDir(String path) {
```

Comment:
I'd rename this to `s/initDir/makeDir`. Everybody knows mkdir so it will ring a bell :)
Or, `createDir`, as besides creation, no other initialization is taking place, so that makes it more explicit.

Reply:
Done.